### ISSUE TITLE
Comparison to other ledger databases could be useful.

### ISSUE THREAD

**Issue by lukaszlew** [Remi-Coulom/joedb#9]
There are other ledger databases (most famously https://aws.amazon.com/qldb/), it would be interesting to see a comparison in README. 

(It seems that joedb is a ledger DB, but I'm not sure.)

**Comment by Remi-Coulom** [comment]
Thanks for your feedback. I may add something to the introduction.

qldb is a blockchain,  "cryptographically verifiable" log. There is no such thing at all in joedb.

Joedb is more like a minimalist relational database, without SQL.  Instead of SQL, you get a C++ in-memory data structures that you can manipulate like containers (in that way, it may be a bit similar to https://redis.io/). In terms of history and distributed concurrency, it is "git for structured data without merging" (linear history only).

There is very little information in this README. The intro there has more:
https://www.remi-coulom.fr/joedb/intro.html
I should probably copy the whole intro into the README.

I use joedb as the data format for my deep learning software. For storing neural networks, training sets, and managing distributed training. It can manipulate vectors of primitive types very efficiently.

For neural networks, most other tools use Google's protobuf. Joedb's "embedded" features are very similar to protobuf, without the 2Gb limit. I can also store blobs for images, which are not loaded into memory, so it is no problem to have a 1Tb joedb database for the training set.

In addition to protobuf-like features, joedb supports concurrent access, with ACID transactions, and git-like distributed operation. So I can very easily and efficiently push and pull new pictures or manually fixed labels into the training set for all my machines. The distributed features are also very useful to handle synchronization of multiple machines running self-play games for my alphazero framework.

As far as I know, nobody but me is using joedb anyway, and I do not have much motivation to advertise it. It is an extremely convenient tool, though, and I am not aware of any alternative way to do what it does with this much convenience, performance, and simplicity.
